src/modules/dashboard/components/ActionsPageFeed/ActionsPageFeedItemWithIPFS/ActionsPageFeedItemWithIPFS.tsx

```diff
@@ -34,8 +54,33 @@ const ActionsPageFeedItemWithIPFS = ({
     return undefined;
   }, [annotation, ipfsDataJSON]);
 
-  const annotationMessage = getAnnotationMessage();
+  const location = useLocation();
+  // trouble connecting to IPFS
+  if (!ipfsDataJSON) {
+    return (
+      <CalloutCard
+        label={MSG.warningTitle}
+        labelValues={{
+          span: (chunks) => (
+            <span className={styles.noIPFSLabel}>{chunks}</span>
+          ),
+        }}
+        description={MSG.warningText}
+        descriptionValues={{
+          link: (
+            <Link
+              to={location.pathname}
+              text={MSG.reloadLink}
+              className={styles.link}
+              onClick={() => window.location.reload()}
+            />
+          ),
+        }}
+      />
+    );
+  }
 
+  const annotationMessage = getAnnotationMessage();
```

Comment:
Maybe add a spinner loader while trying to fetch / parse the IPFS hash _(and subsequent object)_